File: Backend (Springboot)/Annapurna/src/main/java/com/annapurna/domain/Address.java
package com.annapurna.domain;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String street;
    private String city;
    private String state;
    private String pinCode;
    private String country;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;
}

File: Backend (DNET)/Annapurna/Models/Entities/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Annapurna.Models.Entities
{
    public class Address
    {
        [Key]
        public int Id { get; set; } // changed from long to int

        public string Street { get; set; } = string.Empty;

        public string City { get; set; } = string.Empty;

        public string State { get; set; } = string.Empty;

        public string PinCode { get; set; } = string.Empty;

        public string Country { get; set; } = string.Empty;

        [ForeignKey("User")]
        public int UserId { get; set; } // changed from long to int

        public User User { get; set; } = null!;
    }
}
